app/src/main/java/io/github/zadam/triliumsender/LoginActivity.kt

```diff
@@ -42,129 +41,146 @@ class LoginActivity : AppCompatActivity() {
+                } else {
+                    throw RuntimeException("Unknown code: " + loginResult.errorCode)
+                }
+            }
+        } // uiScope.launch {}
+    } // private fun AttemptLogin
 
+
+    /**
+     * A result from a login attempt.
+     */
+    inner class LoginResult(val success: Boolean, val errorCode: Int?,
+                            val token: String? = null)
+
+    /**
+     * Makes the actual login http request in the IO thread, to avoid blocking the UI thread.
+     *
+     * @return A loginResult object.
+     */
+    private suspend fun doLogin(triliumAddress: String, username: String, password: String): LoginResult {
+        return withContext(Dispatchers.IO) {
+            val tag = "UserLoginCoroutine"
             val client = OkHttpClient()
 
             val json = JSONObject()
             json.put("username", username)
             json.put("password", password)
 
-            val body = RequestBody.create(Utils.JSON, json.toString())
+            val body = json.toString().toRequestBody(Utils.JSON)
             val request = Request.Builder()
-                    .url(triliumAddress + "/api/sender/login")
+                    .url("$triliumAddress/api/sender/login")
                     .post(body)
                     .build()
 
-            val response: Response;
+            val response: Response
 
             try {
+                // In the Dispatchers.IO context, blocking http requests are allowed.
+                @Suppress("BlockingMethodInNonBlockingContext")
                 response = client.newCall(request).execute()
-            }
-            catch (e: Exception) {
-                Log.e(TAG, "Can't connect to Trilium server", e);
+            } catch (e: Exception) {
+                Log.e(tag, "Can't connect to Trilium server", e)
 
-                return LoginResult(false, R.string.error_network_error)
+                return@withContext LoginResult(false, R.string.error_network_error)
             }
 
-            Log.i(TAG,"Response code: " + response.code())
+            Log.i(tag, "Response code: " + response.code)
 
-            if (response.code() == 401) {
-                return LoginResult(false, R.string.error_incorrect_credentials)
+            if (response.code == 401) {
+                return@withContext LoginResult(false, R.string.error_incorrect_credentials)
+            } else if (response.code != 200) {
+                return@withContext LoginResult(false, R.string.error_unexpected_response)
             }
-            else if (response.code() != 200) {
-                return LoginResult(false, R.string.error_unexpected_response)
-            }
-
-            val responseText = response.body()?.string()
 
-            Log.i(TAG,"Response text: " + responseText)
+            // In the Dispatchers.IO context, blocking tasks are allowed.
+            @Suppress("BlockingMethodInNonBlockingContext")
+            val responseText = response.body?.string()
```

Comment:
specifically, this lint is complaining about the string() call, since getting the response body in this way causes the underlying connection to close, see https://square.github.io/okhttp/4.x/okhttp/okhttp3/-response-body/#the-response-body-must-be-closed